docs/topics/compatibility-guide-18.md

```diff
@@ -0,0 +1,623 @@
+>   the corresponding type parameters of the aliased type
+> - \>=1.10: raise the warning to an error
+
+### Prohibit using a type parameter declared for an extension property inside delegate
+
+> **Issue**: [KT-24643](https://youtrack.jetbrains.com/issue/KT-24643)
+>
+> **Component**: Core language
+>
+> **Incompatible change type**: source
+>
+> **Short summary**: Kotlin will prohibit delegating extension properties on a generic type
+> to generic types that use the type parameter of the receiver in an unsafe way
+>
+> **Deprecation cycle**:
+>
+> - 1.6.0: report a warning (or an error in the progressive mode) when delegating an extension property to a type 
+>   that uses type parameters inferred from the delegated property type arguments in particular way
+> - 1.8.0: raise the warning to an error,
+>  `-XXLanguage:-ForbidUsingExtensionPropertyTypeParameterInDelegate` can be used to temporarily revert to the pre-1.8 behavior
+
+### Forbid `@Synchronized` annotation on suspend functions
+
+> **Issue**: [KT-48516](https://youtrack.jetbrains.com/issue/KT-48516)
+>
+> **Component**: Core language
+>
+> **Incompatible change type**: source
+>
+> **Short summary**: Kotlin 1.8 will prohibit placing `@Synchronized` annotation on suspend functions 
+> because a suspension call should not be allowed to happen inside a synchronized block
+>
+> **Deprecation cycle**:
+>
+> - 1.6.0: report a warning on suspend functions annotated with `@Synchronized` annotation,
+    the warning is reported as an error in the progressive mode
+> - 1.8.0: raise the warning to an error,
+    `-XXLanguage:-SynchronizedSuspendError` can be used to temporarily revert to the pre-1.8 behavior
+
+### Prohibit using spread operator for passing arguments to non-vararg parameters 
+
+> **Issue**: [KT-48162](https://youtrack.jetbrains.com/issue/KT-48162)
+>
+> **Component**: Core language
+>
+> **Incompatible change type**: source
+>
+> **Short summary**: Kotlin allowed passing arrays with the spread operator (`*`) to non-vararg array parameters 
+> in certain conditions. Since Kotlin 1.8 this will be prohibited
+>
+> **Deprecation cycle**:
+>
+> - 1.6.0: report a warning (or an error in the progressive mode) on using the spread operator where a non-vararg 
+>   array parameter is expected
+> - 1.8.0: raise the warning to an error,
+>   `-XXLanguage:-ReportNonVarargSpreadOnGenericCalls` can be used to temporarily revert to the pre-1.8 behavior
+
+### Prohibit null-safety violation in lambdas passed to functions overloaded by lambda return type
+
+> **Issue**: [KT-49658](https://youtrack.jetbrains.com/issue/KT-49658)
+>
+> **Component**: Core language
+>
+> **Incompatible change type**: source
+>
+> **Short summary**: Kotlin 1.8 will prohibit returning `null` from lambdas passed to functions overloaded by lambda return type
+> when no overload allows a nullable return type.
```

Comment:
```suggestion
> when no overloading allows a nullable return type.
```

Reply:
An overload is a non here. If its unclear, an "overloaded function" can be used

Reply:
Could you please recheck the new variant? And the [comment](https://github.com/JetBrains/kotlin-web-site/pull/3331/files#r1057343058) above.

Reply:
Not exact enoungh. There are multiple overloads in this situation, it's important that none of them allows that

Reply:
`Kotlin 1.8 will prohibit returning `null` from lambdas passed to functions overloaded by those lambdas' return types when overloads don't allow a nullable return type. `

What's about this?

Reply:
Or

> Kotlin 1.8 will prohibit returning null from lambdas passed to functions when such a function is overloaded by a lambda's return type, and these overloads don't allow a nullable return type.

Reply:
ok, I'll take the first one